src/app/recipes/recipe-detail/recipe-detail.component.ts

```diff
@@ -27,18 +27,17 @@ export class RecipeDetailComponent implements OnInit, OnDestroy {
     this.route.params
       .pipe(
         map((params) => {
-          return +params['id'];
+          return params['id'];
         }),
         switchMap((id) => {
           this.id = id;
-          return this.store;
-        }),
-        select('recipes'),
-        map((recipesState) =>
-          recipesState.recipes.find((_, i) => {
-            return i === this.id;
-          })
-        )
+          return this.store.pipe(select('recipes'));
+        }),    
+        map((recipesState) => {
+          return recipesState.recipes.find((recipe) => {
+            return recipe._id === this.id;
+          });
+        })
```

Comment:
Try to create `selector` with `id` param

Reply:
created already